metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -2114,9 +2118,11 @@ class MetalsLspService(
       change
     }
 
-  private def maybeQuickConnectToBuildServer(
+  private def onBuildTargetChanges(
       params: b.DidChangeBuildTarget
   ): Unit = {
+    // Make sure that no compilation is running, if it is it might not get completed properly
```

Comment:
How exactly?

Reply:
I am not sure, I could not reproduce it reliably. My guess is that when you send one compile first and then change using directives in another compile requests a short while later, last one will get cancelled and the first one forgotten.

Reply:
And get stuck in the queue until next cancel.

Reply:
So it's a `BatchedFunction` thing not a `scala-cli` thing?

Reply:
yes, but it showed up here multiple times when testing.

Reply:
Yeah, it seems Scala CLI doesn't always send the respond, looks like it's a race condition, but might be hard to fix.